src/api/object.rs

```diff
@@ -0,0 +1,132 @@
+use crate::{
+    api::metadata::{ListMeta, Metadata},
+    ErrorResponse,
+};
+use serde::Deserialize;
+use std::fmt::Debug;
+
+/// A raw event returned from a watch query
+///
+/// Note that a watch query returns many of these as newline separated json.
+#[derive(Deserialize, Serialize, Clone)]
+#[serde(tag = "type", content = "object", rename_all = "UPPERCASE")]
+pub enum WatchEvent<K>
+where
+    K: Clone + Metadata,
+{
+    Added(K),
+    Modified(K),
+    Deleted(K),
+    Error(ErrorResponse),
```

Comment:
This wasn't really added here, but.. maybe the error conditions should be moved out into a `Result<WatchEvent<K>>` instead? This should probably be a separate issue rather than blocking this PR.

Reply:
Yeah, that might be something worth raising. Have also considered picking up the upstream WatchEvent from k8s_openapi which in its latest version is very similar to the one we have (although with even more error separation).